numpyro/distributions/distribution.py

```diff
@@ -797,11 +798,13 @@ def rsample(self, key, sample_shape=()):
 
     @property
     def support(self):
-        domain = self.base_dist.support
-        for t in self.transforms:
-            t.domain = domain
-            domain = t.codomain
-        return domain
+        codomain = self.transforms[-1].codomain
+        codomain_event_dim = codomain.event_dim
+        assert self.event_dim >= codomain_event_dim
+        if self.event_dim == codomain_event_dim:
+            return codomain
+        else:
+            return independent(codomain, self.event_dim - codomain_event_dim)
```

Comment:
I believe this needs more complex logic as in https://github.com/pytorch/pytorch/pull/50581/files#diff-8b108de0fae81bfc4f939e40322885653fc3c957b2f8ad0ff25aafc16bf2c639

Reply:
Yes, I added a comment there. Here we fixed `self.event_dim` directly from the constructor.

Reply:
Ok, it seems reasonable to defer `.forward_shape()` to a separate PR.